### PR TITLE
Patch/copywriting

### PR DESCRIPTION
**Pull request by lyqht**
# Changelog

## Chapter 1

- refactor `retrieveNotes` to be more readable
- grammar, spelling and rephrasing edits
- added link to “The basic template for a Node.js project with Express and Pug”
- added bash command to inspect tar
- added example link + screenshot of container image uploaded on dockerhub
- added instructions to address open issue #4 

## Chapter 2

- added custom graphics
    - diagram for Kubernetes resource categories overview
    - diagram for what is a Pod
- grammar, spelling and rephrasing edits
- Under “Defining a Service” section,
    - removed a duplicated part on “where you should save the  definition as it is a best practice” → it is already mentioned before.
    - felt that the 3 dash separation instructions to be a bit confusing, so I rewrote that portion significantly.

## Chapter 3

- Link for “The application that you're about to scale is already deployed in the cluster” was pointing to "Developing and deploying Spring Boot microservices on Kubernetes” guide. Changed this to point to Chapter 2 link instead.
- grammar, spelling and rephrasing edits

## Chapter 4

- grammar, spelling and rephrasing edits
- added screenshot of eks aws console having the cluster created
- added section on installing aws-cli to verify that the user is authenticated and able to get eks clusters
- rephrased section on calculating the number of knote pods that can be spurned up

---

# Notes to Project maintainers:

- In Chapter 2 onwards, we have been telling readers that the application is available from Docker as:
    - `learnk8s/knote-js:1.0.0` or
    - `learnk8s/knote-js:2.0.0`
    
    → it is up to y’all if you folks want to release `1.0.1` & `2.0.1` containing the dependency updates, or replace the existing version.
- **Please check the following in Chapter 2**
    - the content of `assets/service.svg` is the same as  `assets/service-with-ports.svg` → from reading the article, it feels that it is intended to be different.
    - the pro tip section on *`kubectl explain service.spec.type`* is cut off a little, so I changed that from an inline code to block code. please help to check if later it still renders as intended.
- **Comments on Chapter 3**
    - From what I understand, the article’s intent was for teaching the importance of statelessness and the concepts of Scaling in Kubernetes. I think the article delivered these 2 intents very well, but *the frequent mentions of how the reader has learnt how to refactor the app to include MinIO was a bit confusing for me*.
        - As a reader, I felt that I didn’t really learn much about the refactoring portion, since *I’m provided a container image that already contains the refactored code*.
        - While I could select & compare files from Folder 03 to Folder 04 using VSC, I’m not sure if most readers would know how to do this.
        - Even if the reader knows to select & compare like this, if they have not worked with storage buckets, they may not understand concepts like bucket, buffer, as well as why the changes in the multer option from a static folder path to memory storage & the new endpoint is required.
        - While we could include a short explanation on the changes, this may prolong the article unnecessarily.
    - **I suggest that we don’t emphasize on the fact that the user has learnt how to refactor the code**. We could phrase the scenario in the way that they have a teammate who has already helped to refactor the application code, but the reader has to know how to scale the app for the team.

### CODE REVIEW

File: 02_deploying_to_kubernetes.md [hunk 1]
```diff
@@ -269,7 +271,11 @@ kubectl explain deployment.spec.replicas
 
 **Now that you know how to look up the documentation of Kubernetes resources, let's turn back to the Deployment.**
 
-The first four lines define the type of resource (Deployment), the version of this resource type (`apps/v1`), and the name of this specific resource (`knote`):
+The first four lines define: 
+
+- the type of resource (Deployment)
+- the version of this resource type (`apps/v1`), and 
+- the name of this specific resource (`knote`)
```
Comment (danielepolencic) [inline 02_deploying_to_kubernetes.md:278]: In general, for bullet points:

- The first letter is a capital letter.
- The bullet point ends with a dot.

File: 02_deploying_to_kubernetes.md [hunk 2]
```diff
@@ -436,9 +442,9 @@ A Service forwards requests to a set of Pods:
 
 In this regard, a Service is akin to a load balancer.
 
-Here is the definition of a Service that makes your Knote Pod accessible from outside the cluster:
+Now, let's add a Service resource definition to make your Knote Pod accessible from outside the cluster. We will add this resource definition to the existing YAML file:
```
Comment (danielepolencic) [inline 02_deploying_to_kubernetes.md:445]: One sentence per paragraph.

File: 02_deploying_to_kubernetes.md [hunk 3]
```diff
@@ -550,7 +568,10 @@ In this case, the type is `LoadBalancer`, which makes the exposed Pods accessibl
 
 The default Service type is `ClusterIP`, which makes the exposed Pods only accessible from within the cluster.
 
-> **Pro tip:** find out about all available Service types with `kubectl explain service.spec.type`.
+> **Pro tip:** find out about all available Service types with
+> ```
+> kubectl explain service.spec.type
+> ```
```
Comment (danielepolencic) [inline 02_deploying_to_kubernetes.md:574]: This is a limitation of our markdown engine, but the blockquote will break with the current change as the triple ticks are rendered as a full editor.

File: 03_scaling.md [hunk 4]
```diff
@@ -330,13 +327,9 @@ spec:
             claimName: minio-pvc
 ```
 
-Go on and save the content in a file named `minio.yaml` in the `kube` directory.
-
-The YAML file has a Deployment, Service and PersistenVolumeClaim definition.
+Just like the YAML file `kube/mongo.yaml`, this file also has a Deployment, Service and PersistenVolumeClaim resource definition. The same concepts and understanding can be applied here.
```
Comment (danielepolencic) [inline 03_scaling.md:330]: One sentence per paragraph.

File: 04_deploying_to_the_cloud.md [hunk 5]
```diff
@@ -111,6 +111,28 @@ aws_access_key_id=[access-key-id]
 aws_secret_access_key=[secret-access-key]
 ```
 
+_Remember to include the `[default]` in the file!_
+
+Now, you can [install the aws-cli](https://github.com/aws/aws-cli)
+
+```
```
Comment (danielepolencic) [inline 04_deploying_to_the_cloud.md:118]: this should be  tripe ticks + `terminal|command=1|title=bash`

File: 04_deploying_to_the_cloud.md [hunk 6]
```diff
@@ -111,6 +111,28 @@ aws_access_key_id=[access-key-id]
...
+sudo python -m pip install awscli
+```
+
+
+If you are authenticated, when you run the command below
+```
```
Comment (danielepolencic) [inline 04_deploying_to_the_cloud.md:124]: this should be  tripe ticks + `terminal|command=1|title=bash`

File: 04_deploying_to_the_cloud.md [hunk 7]
```diff
@@ -111,6 +111,28 @@ aws_access_key_id=[access-key-id]
...
+aws eks list-clusters --region=us-east-1
+```
+
+You should get a valid response as such if you have not created any clusters before.
+
+```
```
Comment (danielepolencic) [inline 04_deploying_to_the_cloud.md:130]: I think the output can be included in the previous snippet. The markdown engine will parse the output accordingly.

File: 02_deploying_to_kubernetes.md [hunk 8]
```diff
@@ -149,13 +149,15 @@ In other words, you describe how you want the deployment of your application to
 
 The "language" that you use to communicate with Kubernetes consists of so-called Kubernetes resources.
 
-There are many different Kubernetes resources — each is responsible for a specific aspect of your application.
+There are many different Kubernetes resources — each is responsible for a specific aspect of your application. Here is a high level overview of the 5 categories of Kubernetes resources.
 
 > You can find the full list of Kubernetes resources in the [Kubernetes API reference](https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.20/).
 
-Kubernetes resources are defined in YAML files and submitted to the cluster through the Kubernetes HTTP API.
+![](assets/kubernetes-resources-overview.jpeg)
```
[the remark this answers is not in the document]
Comment (lyqht) [reply 02_deploying_to_kubernetes.md:156]: Ah I wanted to show at a high level the type of resources that are available. The resource and description are based on what is given at the Kubernetes API reference. If this is confusing, we can remove the diagram instead!

**Pull request merged**